File: ze-studio/send-telegram.py
#!/usr/bin/env python3
"""
Telegram Bot Integration - Python Backend
Отправка заявок из контактной формы в Telegram
"""

from http.server import HTTPServer, BaseHTTPRequestHandler
import json
import urllib.request
import urllib.parse
from datetime import datetime

# ============================================
# CONFIGURATION
# ============================================
TELEGRAM_BOT_TOKEN = '8223297194:AAGrxlYqx67E4FqjMCd_3m_BWNAD-Lxd59A'
TELEGRAM_CHAT_ID = '8141249044'  # Владислав (@I9006)

# ============================================
# FORMAT MESSAGE
# ============================================
def format_telegram_message(data):
    """Форматирование сообщения для Telegram"""
    goal = data.get('goal', 'Новая заявка')
    
    message = f"🎯 <b>{goal}</b>\n\n"
    
    # Basic info
    if data.get('name'):
        message += f"👤 <b>Имя:</b> {data['name']}\n"
    
    if data.get('phone'):
        message += f"📱 <b>Телефон:</b> {data['phone']}\n"
    
    if data.get('car'):
        message += f"🚗 <b>Автомобиль:</b> {data['car']}\n"
    
    # Service details
    if data.get('service'):
        message += f"\n💼 <b>Услуга:</b> {data['service']}\n"
    
    if data.get('services'):
        services_list = data['services']
        if isinstance(services_list, list) and services_list:
            message += f"\n💼 <b>Услуги:</b> {', '.join(services_list)}\n"
        elif services_list:
            message += f"\n💼 <b>Услуги:</b> {services_list}\n"
    
    if data.get('old_film'):
        message += f"🔄 <b>Старая пленка:</b> {data['old_film']}\n"
    
    if data.get('film'):
        message += f"🎞 <b>Пленка:</b> {data['film']}\n"
    
    if data.get('timing'):
        message += f"⏰ <b>Когда:</b> {data['timing']}\n"
    
    if data.get('when'):
        message += f"⏰ <b>Когда:</b> {data['when']}\n"
    
    if data.get('source'):
        message += f"📍 <b>Откуда узнали:</b> {data['source']}\n"
    
    if data.get('has_photo'):
        message += f"📷 <b>Есть фото:</b> {data['has_photo']}\n"
    
    if data.get('contact_method'):
        message += f"📞 <b>Как связаться:</b> {data['contact_method']}\n"
    
    # Comments
    if data.get('comment'):
        message += f"\n💬 <b>Комментарий:</b>\n{data['comment']}\n"
    
    if data.get('notes'):
        message += f"\n📝 <b>Уточнения:</b>\n{data['notes']}\n"
    
    if data.get('message'):
        message += f"\n✉️ <b>Сообщение:</b>\n{data['message']}\n"
    
    # Timestamp
    if data.get('timestamp'):
        message += f"\n🕐 {data['timestamp']}"
    else:
        message += f"\n🕐 {datetime.now().strftime('%d.%m.%Y %H:%M:%S')}"
    
    return message

# ============================================
# SEND TO TELEGRAM
# ============================================
def send_telegram_message(message, chat_id, bot_token):
    """Отправка сообщения в Telegram"""
    url = f"https://api.telegram.org/bot{bot_token}/sendMessage"
    
    data = {
        'chat_id': chat_id,
        'text': message,
        'parse_mode': 'HTML'
    }
    
    try:
        req = urllib.request.Request(
            url,
            data=urllib.parse.urlencode(data).encode('utf-8'),
            headers={'Content-Type': 'application/x-www-form-urlencoded'}
        )
        
        with urllib.request.urlopen(req, timeout=10) as response:
            result = json.loads(response.read().decode('utf-8'))
            
            if result.get('ok'):
                return {
                    'success': True,
                    'message_id': result['result']['message_id']
                }
            else:
                return {
                    'success': False,
                    'error': result.get('description', 'Unknown error')
                }
    
    except Exception as e:
        return {
            'success': False,
            'error': str(e)
        }

# ============================================
# HTTP REQUEST HANDLER
# ============================================
class TelegramHandler(BaseHTTPRequestHandler):
    
    def do_OPTIONS(self):
        """Handle preflight CORS"""
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
    
    def do_POST(self):
        """Handle POST request"""
        try:
            # Read request body
            content_length = int(self.headers.get('Content-Length', 0))
            body = self.rfile.read(content_length).decode('utf-8')
            
            # Parse JSON
            data = json.loads(body)
            
            # Validate required fields
            if not data.get('name') or not data.get('phone'):
                self.send_response(400)
                self.send_header('Access-Control-Allow-Origin', '*')
                self.send_header('Content-Type', 'application/json; charset=utf-8')
                self.end_headers()
                self.wfile.write(json.dumps({
                    'success': False,
                    'error': 'Missing required fields: name or phone'
                }).encode('utf-8'))
                return
            
            # Format message
            message = format_telegram_message(data)
            
            # Send to Telegram
            result = send_telegram_message(message, TELEGRAM_CHAT_ID, TELEGRAM_BOT_TOKEN)
            
            if result['success']:
                self.send_response(200)
                self.send_header('Access-Control-Allow-Origin', '*')
                self.send_header('Content-Type', 'application/json; charset=utf-8')
                self.end_headers()
                self.wfile.write(json.dumps({
                    'success': True,
                    'message': 'Заявка успешно отправлена в Telegram',
                    'message_id': result['message_id']
                }).encode('utf-8'))
            else:
                self.send_response(500)
                self.send_header('Access-Control-Allow-Origin', '*')
                self.send_header('Content-Type', 'application/json; charset=utf-8')
                self.end_headers()
                self.wfile.write(json.dumps({
                    'success': False,
                    'error': result['error']
                }).encode('utf-8'))
        
        except json.JSONDecodeError:
            self.send_response(400)
            self.send_header('Access-Control-Allow-Origin', '*')
            self.send_header('Content-Type', 'application/json; charset=utf-8')
            self.end_headers()
            self.wfile.write(json.dumps({
                'success': False,
                'error': 'Invalid JSON'
            }).encode('utf-8'))
        
        except Exception as e:
            self.send_response(500)
            self.send_header('Access-Control-Allow-Origin', '*')
            self.send_header('Content-Type', 'application/json; charset=utf-8')
            self.end_headers()
            self.wfile.write(json.dumps({
                'success': False,
                'error': str(e)
            }).encode('utf-8'))
    
    def log_message(self, format, *args):
        """Custom log format"""
        print(f"[{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] {format % args}")

# ============================================
# MAIN
# ============================================
if __name__ == '__main__':
    PORT = 9001
    server = HTTPServer(('0.0.0.0', PORT), TelegramHandler)
    print(f"🚀 Telegram Backend запущен на порту {PORT}")
    print(f"📱 Bot: @ze_studio48bid_bot")
    print(f"💬 Chat ID: {TELEGRAM_CHAT_ID}")
    print(f"⏳ Ожидание запросов...")
    server.serve_forever()

File: send-telegram.php
<?php
/**
 * Telegram Bot Integration
 * Отправка заявок из контактной формы в Telegram
 */

// Enable error reporting for debugging
error_reporting(E_ALL);
ini_set('display_errors', 1);

// CORS headers
header('Access-Control-Allow-Origin: *');
header('Access-Control-Allow-Methods: POST, OPTIONS');
header('Access-Control-Allow-Headers: Content-Type');
header('Content-Type: application/json; charset=utf-8');

// Handle preflight
if ($_SERVER['REQUEST_METHOD'] === 'OPTIONS') {
    http_response_code(200);
    exit();
}

// Only accept POST
if ($_SERVER['REQUEST_METHOD'] !== 'POST') {
    http_response_code(405);
    echo json_encode(['success' => false, 'error' => 'Method not allowed']);
    exit();
}

// ============================================
// CONFIGURATION
// ============================================
define('TELEGRAM_BOT_TOKEN', 'YOUR_BOT_TOKEN_HERE');
define('TELEGRAM_CHAT_ID', 'YOUR_CHAT_ID_HERE'); // Владислав (@I9006)

// ============================================
// GET INPUT DATA
// ============================================
$input = file_get_contents('php://input');
$data = json_decode($input, true);

if (!$data) {
    http_response_code(400);
    echo json_encode(['success' => false, 'error' => 'Invalid JSON']);
    exit();
}

// ============================================
// VALIDATE REQUIRED FIELDS
// ============================================
$requiredFields = ['name', 'phone'];
foreach ($requiredFields as $field) {
    if (empty($data[$field])) {
        http_response_code(400);
        echo json_encode(['success' => false, 'error' => "Missing required field: $field"]);
        exit();
    }
}

// ============================================
// FORMAT MESSAGE
// ============================================
function formatTelegramMessage($data) {
    $goal = $data['goal'] ?? 'Новая заявка';
    
    $message = "🎯 <b>$goal</b>\n\n";
    
    // Basic info
    if (!empty($data['name'])) {
        $message .= "👤 <b>Имя:</b> " . htmlspecialchars($data['name']) . "\n";
    }
    
    if (!empty($data['phone'])) {
        $message .= "📱 <b>Телефон:</b> " . htmlspecialchars($data['phone']) . "\n";
    }
    
    if (!empty($data['car'])) {
        $message .= "🚗 <b>Автомобиль:</b> " . htmlspecialchars($data['car']) . "\n";
    }
    
    // Service details
    if (!empty($data['service'])) {
        $message .= "\n💼 <b>Услуга:</b> " . htmlspecialchars($data['service']) . "\n";
    }
    
    if (!empty($data['film'])) {
        $message .= "🎞 <b>Пленка:</b> " . htmlspecialchars($data['film']) . "\n";
    }
    
    if (!empty($data['timing'])) {
        $message .= "⏰ <b>Когда:</b> " . htmlspecialchars($data['timing']) . "\n";
    }
    
    // Comments
    if (!empty($data['comment'])) {
        $message .= "\n💬 <b>Комментарий:</b>\n" . htmlspecialchars($data['comment']) . "\n";
    }
    
    if (!empty($data['notes'])) {
        $message .= "\n📝 <b>Уточнения:</b>\n" . htmlspecialchars($data['notes']) . "\n";
    }
    
    if (!empty($data['message'])) {
        $message .= "\n✉️ <b>Сообщение:</b>\n" . htmlspecialchars($data['message']) . "\n";
    }
    
    // Timestamp
    if (!empty($data['timestamp'])) {
        $message .= "\n🕐 " . htmlspecialchars($data['timestamp']);
    } else {
        $message .= "\n🕐 " . date('d.m.Y H:i:s');
    }
    
    return $message;
}

// ============================================
// SEND TO TELEGRAM
// ============================================
function sendTelegramMessage($message, $chatId, $botToken) {
    $url = "https://api.telegram.org/bot{$botToken}/sendMessage";
    
    $postData = [
        'chat_id' => $chatId,
        'text' => $message,
        'parse_mode' => 'HTML'
    ];
    
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $url);
    curl_setopt($ch, CURLOPT_POST, 1);
    curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($postData));
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, true);
    curl_setopt($ch, CURLOPT_TIMEOUT, 10);
    
    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    $curlError = curl_error($ch);
    curl_close($ch);
    
    if ($curlError) {
        return [
            'success' => false,
            'error' => 'cURL error: ' . $curlError
        ];
    }
    
    $result = json_decode($response, true);
    
    if ($httpCode !== 200 || !$result['ok']) {
        return [
            'success' => false,
            'error' => $result['description'] ?? 'Unknown Telegram API error',
            'http_code' => $httpCode,
            'response' => $result
        ];
    }
    
    return [
        'success' => true,
        'message_id' => $result['result']['message_id']
    ];
}

// ============================================
// MAIN EXECUTION
// ============================================

// Check if chat ID is configured
if (TELEGRAM_CHAT_ID === 'PLACEHOLDER') {
    // Try to get chat ID from recent messages
    $getUpdatesUrl = "https://api.telegram.org/bot" . TELEGRAM_BOT_TOKEN . "/getUpdates";
    $updates = @file_get_contents($getUpdatesUrl);
    
    if ($updates) {
        $updatesData = json_decode($updates, true);
        if (!empty($updatesData['result'])) {
            $lastUpdate = end($updatesData['result']);
            $detectedChatId = $lastUpdate['message']['chat']['id'] ?? null;
            
            if ($detectedChatId) {
                // Use detected chat ID
                $chatId = $detectedChatId;
                
                echo json_encode([
                    'success' => true,
                    'message' => 'Отправлено',
                    'note' => 'Chat ID обнаружен автоматически: ' . $chatId . '. Пожалуйста, обновите TELEGRAM_CHAT_ID в send-telegram.php',
                    'chat_id' => $chatId
                ]);
                exit();
            }
        }
    }
    
    http_response_code(500);
    echo json_encode([
        'success' => false,
        'error' => 'Chat ID не настроен. Пожалуйста, отправьте сообщение боту @ze_studio48bid_bot и обновите эту страницу.'
    ]);
    exit();
}

$message = formatTelegramMessage($data);
$result = sendTelegramMessage($message, TELEGRAM_CHAT_ID, TELEGRAM_BOT_TOKEN);

if ($result['success']) {
    http_response_code(200);
    echo json_encode([
        'success' => true,
        'message' => 'Заявка успешно отправлена в Telegram',
        'message_id' => $result['message_id']
    ]);
} else {
    http_response_code(500);
    echo json_encode([
        'success' => false,
        'error' => $result['error'],
        'details' => $result
    ]);
}
?>
